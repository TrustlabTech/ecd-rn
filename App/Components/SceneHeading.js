/**
 * Early Childhood Development App
 * @copyright 2016 Global Consent Ltd
 * Civvals, 50 Seymour Street, London, England, W1H 7JG
 * @author Werner Roets <werner@io.co.za>
 */

import React, { Component } from 'react'
import {
  View,
  Text
} from 'react-native'
import { FontSizes } from '../GlobalStyles'

export default class SceneHeading extends Component {

  render () {
    return (
      <View style={{alignItems: 'center', marginTop: 8, marginBottom: 5}}>
        <Text style={{fontSize: FontSizes.h4, fontWeight: 'bold'}}>{this.props.text}</Text>
      </View>
    )
  }
}

SceneHeading.propTypes = {
  text: React.PropTypes.string
}
