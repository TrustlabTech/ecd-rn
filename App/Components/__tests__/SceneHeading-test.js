/**
 * Early Childhood Development App
 * @copyright 2016 Global Consent Ltd
 * Civvals, 50 Seymour Street, London, England, W1H 7JG
 * @author Werner Roets <werner@io.co.za>
 */

import 'react-native'
import React from 'react'
import SceneHeading from '../SceneHeading'

import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer.create(
    <SceneHeading text="Test SceneHeading"/>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})