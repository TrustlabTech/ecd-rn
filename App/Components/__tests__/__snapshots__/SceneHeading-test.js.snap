exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "marginBottom": 5,
      "marginTop": 8,
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 26,
        "fontWeight": "bold",
      }
    }>
    Test SceneHeading
  </Text>
</View>
`;
